### DISCUSSION TITLE
No 'after_val_epoch' method in the evaluation hook

### DISCUSSION THREAD

**Discussion by rubeea** [open-mmlab/mmpose#862] [General]
Hi,

As per the documentation of customizing workflows at https://github.com/open-mmlab/mmpose/blob/master/docs/tutorials/6_customize_runtime.md#customize-workflow
There must be a method called "after_val_epoch" that is called is the workflow is set to [('train', 1), ('val', 1)] but I did not see any such method in the evaluation hook (https://github.com/open-mmlab/mmpose/blob/master/mmpose/core/evaluation/eval_hooks.py). Can you kindly provide an implementation of the method? Thanks

**Comment by ly015** [comment]
As mentioned in the document, we recommend using workflow=[('train',1)] and let the EvalHook.after_train_epoch perform validation. If you need 'val' phase in the workflow, you can add an after_val_epoch method to EvalHook just like its after_train_epoch. Alternatively, you can rename the existing after_train_epoch to after_epoch. According to the implementation of the base class [Hook](https://github.com/open-mmlab/mmcv/blob/master/mmcv/runner/hooks/hook.py), after_epoch will be called either after a training or a validation epoch.

**Comment by innerlee** [comment]
@ly015 we may remove val workflow related things from doc, as it causes more trouble than help

**Comment by rubeea** [comment]
> @ly015 we may remove val workflow related things from doc, as it causes more trouble than help

@innerlee if you recommend using the workflow=[('train',1)]  and plan to remove the val workflow how can one check for overfitting? If we see the tensorboard logs only the validation set metrics are printed there not the train set logs.
